setup.py

```diff
@@ -705,11 +712,20 @@ def reformat_run_count(count_str):
     version_number = version_number + local_version
     if is_rocm and enable_rocm_profiling:
         version_number = version_number + ".profiling"
-
+extras_require = {}
 if wheel_name_suffix:
     if not (enable_training and wheel_name_suffix == "gpu"):
         # for training packages, local version is used to indicate device types
         package_name = f"{package_name}-{wheel_name_suffix}"
+    if wheel_name_suffix == "gpu" and cuda_version_major is not None:
+        extras_require = {
+            # Optional 'cuda_dlls' dependencies
+            "cuda_dlls": [
+                f"nvidia-cudnn-cu{cuda_version_major}",
```

Comment:
Not sure whether the following are needed:
nvidia-nvjitlink-cu12 ?

Reply:
it came within nvidia-cufft-cu1x

Reply:
See this: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/9902

We probably should not depend on nvjitlink

Reply:
For your reference, our Linux CUDA 12 python package depends on the following CUDA libs:

1.         libcublas.so.12
2.         libcublasLt.so.12
3.         libcudart.so.12
4.         libcudnn.so.9
5.         libcudnn_adv.so.9
6.         libcudnn_cnn.so.9
7.         libcudnn_engines_precompiled.so.9
8.         libcudnn_engines_runtime_compiled.so.9
9.         libcudnn_graph.so.9
10.         libcudnn_heuristic.so.9
11.         libcudnn_ops.so.9
12.         libcufft.so.11
13.         libcurand.so.10
14.         libnvJitLink.so.12
15.         libnvrtc.so.12

Reply:
nvidia-cufft-cu12 depends on nvidia-nvjitlink-cu12. So they will get installed together.

Reply:
It's fine to add them even when one depends on another. For example, here is the pytorch configuration:

https://github.com/pytorch/pytorch/blob/889717aabdfd48de0086309b57310792cf2e2ad6/.github/scripts/generate_binary_build_matrix.py#L71-L83

Reply:
nvidia-cublas-cu12 come with cudnn.
I am adding nvidia-cuda-runtime-cu12 and vidia-curand-cu12